middleware/proxy_test.go

```diff
@@ -89,6 +100,16 @@ func TestProxy(t *testing.T) {
 	body = rec.Body.String()
 	assert.Equal(t, "target 2", body)
 
+	// Provider
```

Comment:
please create separate function for this testcase. These tests (functions) that contain actually multiple testcases that sometimes reuse objects are bad practice. If test cases are isolated it is easier to test them separately and overall have less sideeffects.